### PR TITLE
FEATURE: Allow to edit roles assigned to users

### PR DESCRIPTION
**Pull request by kdambekalns**
This allows to assign roles to frontend users. All roles that have
the Flowpack.FrontendLogin:User role as their (indirect) parent can be
assigned.

### CODE REVIEW

File: Classes/Controller/ModuleController.php [hunk 1]
```diff
@@ -114,14 +122,17 @@ public function newAction(User $user = null): void
...
      * @return void
      * @Flow\Validate(argumentName="username", type="\Neos\Flow\Validation\Validator\NotEmptyValidator")
      * @Flow\Validate(argumentName="username", type="\Neos\Neos\Validation\Validator\UserDoesNotExistValidator")
      * @Flow\Validate(argumentName="password", type="\Neos\Neos\Validation\Validator\PasswordValidator", options={ "allowEmpty"=0, "minimum"=1, "maximum"=255 })
      */
-    public function createAction($username, array $password, User $user, ?\DateTime $expirationDate): void
+    public function createAction($username, array $password, User $user, ?\DateTime $expirationDate, array $roleIdentifiers = []): void
     {
-        $user = $this->userService->addUser($username, $password[0], $user, [self::$roleIdentifier], self::$authenticationProviderName);
+        // make sure self::$roleIdentifier is always added
```
Comment (bwaidelich) [inline Classes/Controller/ModuleController.php:132]: possible privilege escalation? i.e. are the given $roleIdentifiers checked against allowed roles?

File: Classes/Controller/ModuleController.php [hunk 2]
```diff
@@ -220,16 +233,20 @@ public function editAccountAction(Account $account): void
...
      * @return void
      * @throws NeosDomainException
      * @throws IllegalObjectTypeException
      * @throws UnsupportedRequestTypeException
      */
-    public function updateAccountAction(Account $account, array $password = []): void
+    public function updateAccountAction(Account $account, array $roleIdentifiers = [], array $password = []): void
     {
         if ($this->checkAccount($account)) {
+            // make sure self::$roleIdentifier is always added
```
Comment (bwaidelich) [inline Classes/Controller/ModuleController.php:249]: same here (?)

File: Classes/Controller/ModuleController.php (hunk 1 above)
Comment (kdambekalns) [reply Classes/Controller/ModuleController.php:132]: Now only "FE user roles" can be assigned.

File: Classes/Controller/ModuleController.php (hunk 2 above)
Comment (kdambekalns) [reply Classes/Controller/ModuleController.php:249]: Now only "FE user roles" can be assigned/removed. Possibly existing other roles are kept unchanged.

**Review by kdambekalns — COMMENTED** [review]
@bwaidelich I hope that does the job to your satisfaction. I did not add further checks for the user having (at least) the assigned roles, since this is unlikely: backend users will likely not have frontend roles anyway…

**Comment by kdambekalns** [comment]
I know I have been silent on this or a while, but it would still be great to have it reviewed and possibly merged…

**Review by bwaidelich — APPROVED** [review]
Looks good (by reading)
nice feature, thanks for taking care

**Pull request merged**